.github/workflows/enos-run.yml

```diff
@@ -157,9 +156,9 @@ jobs:
           ENOS_VAR_skip_failing_bats_tests: "true"
         run: |
           export ENOS_VAR_enos_user=$GITHUB_ACTOR && \
-          enos scenario run --timeout 60m0s --chdir ./enos integration test:${{ matrix.test}} builder:crt
+          enos scenario run --timeout 60m0s --chdir ./enos ${{ matrix.filter}} builder:crt
       - name: Destroy Enos scenario
-        if: ${{ always() }}
+        if: ${{ failure() && steps.run.conclusion == 'failure' || failure() && steps.run_retry.conclusion == 'failure' }}
```

Comment:
It seems even though the destroy step failed before it did not error out in GHA with Enos version 0.0.10 and before, but now failure in destroy step does return error with Enos 0.0.11 .  May be we can add `continue-on-error: true` as a temporary workaround to this step with a comment, as the failure of destroy step does not determine the failure of Enos run

Reply:
(Not replying to Jaymala here. We talked about this in Slack. This comment is for posterity.)

In the case of continue-on-error, there will still be an icon indicating this step failed if the runs were successful. We know that's okay, but that isn't immediately clear to a random user that may see this.

continue-on-error is present for the whole job which already has this suggested functionality present, but my desire is to not have the red X indicating failure on this step if the step didn't need to run in the first place.

This solution keeps the step from running when it doesn't need to run and shows that it was appropriately skipped. This allows the user to just glance at it and not need to investigate further or have tribal knowledge about the workflow.

Reply:
I'm a little confused.

The previous steps are executing `enos scenario run` which, based on my understanding, includes a `destroy`. Previously, this step executes another destroy. Now, this change is suggesting to only run this destroy if `run` or `run_retry` failed (whether it failed when running or destroying? Is that correct?

I'm not clear on what `continue-on-error` would do here.

Reply:
Addendum to the last bit, for posterity: Jaymala suggested that this could and probably should be handled in Enos itself. Terraform _also_ is okay with saying "There is nothing to destroy. My job is done." I'm inclined to agree that a destroy that does not need to destroy anything should exit 0. Hopefully this is temporary, and we will add functionality in Enos to allow this to execute without condition.

Reply:
@moduli That's correct. The destroy part of the lifecycle of `enos scenario run` only executes if the apply part of it was successful. This discrete call to `enos scenario destroy` ensures that the infrastructure is always destroyed even in the case of the run or run-retry step's failure.

With the previous logic, if the destroy step ran after the run was successful, you would have an icon on the workflow run page that indicates the failure of a step. This change makes it to where the step is explicitly skipped in the event that the run (or retry of course) step was successful.

But, because of the discussion in this PR, Ryan is making a change to Enos that will make the `enos scenario destroy` action not return a non-zero exit code if it simply has nothing to destroy. As a result, we will likely move the conditional from this step before it is merged.

Reply:
Ah. Got it.

Reply:
Removing the condition as Ryan has a change to Enos that will let destroy run and exit cleanly if it has nothing to destroy.

Reply:
```suggestion
```